components/x-gift-article/src/v2/GiftLinkSection.jsx

```diff
@@ -9,13 +9,16 @@ export const GiftLinkSection = (props) => {
 	const createLinkHandler = async () => {
 		switch (shareType) {
 			case ShareType.gift:
-				await actions.createGiftUrl()
+				actions.createGiftUrl()
+				actions.initOShare('#social-share-buttons')
 				break
 			case ShareType.nonGift:
-				await actions.shortenNonGiftUrl()
+				actions.shortenNonGiftUrl()
+				actions.initOShare('#social-share-buttons')
 				break
 			case ShareType.enterprise:
-				await actions.createEnterpriseUrl()
+				actions.createEnterpriseUrl()
+				actions.initOShare('#social-share-buttons')
 				break
 			default:
```

Comment:
I don't think this is really correct because you removed the `await`, so we are still not guaranteeing to call initOShare after the updater rendered the share component

Reply:
please see https://github.com/Financial-Times/x-dash/compare/d7a809a697d647d358eb7b777fd3961742e13392..5605a9afdb6578319e8b16c6909aebe0fd77b64c